lib/monocle.js

```diff
@@ -20,68 +23,119 @@ Monocle.prototype.getCache = function() {
+        body: batched
+    }).then(function(results) {
+        results.forEach(function(result, i) {
+            if (result.status >= 200 && result.status < 300) {
+                batched[i].resolve(result.body);
+            } else {
+                batched[i].reject(result.body);
+            }
+        })
+    }).catch(function(error) {
+        batched.forEach(function(apiCall, i) {
+            batched[i].reject(error);
+        });
+    });
+};
 
-    var cacheResource = function(method, resource) {
-        if ('get' === method && resource.$id && resource.$expires) {
-            this._cache.put(resource.$id, resource, resource.$expires);
-        }
-        return resource;
-    };
+var _handle = function(method, path, options) {
+    switch (method) {
+        // Check cache if attempting to get resource
+        case 'get':
+            var cached = this._cache.get(path);
+            if (cached) {
+                if (!options || !options.props) {
+                    return Promise.resolve(cached);
+                }
+                var found = true;
+                for (var i in options.props) {
+                    if (!cached.hasOwnProperty(options.props[i])) {
+                        this._cache.remove(path);
+                        found = false;
+                        break;
+                    }
+                }
+                if (found) return Promise.resolve(cached);
+            }
+            break;
+
+        // Remove from cache when resource is being updated or removed
+        case 'post':
+        case 'put':
+        case 'delete':
+        case 'patch':
+            this._cache.remove(path);
+            break;
+    }
+
+    return (new Promise(function(resolve, reject) {
+        this._batched.push({
+            method: method,
+            path: path,
+            options: options,
+            resolve: resolve,
+            reject: reject
+        });
+        updateBatchTimeout.call(this);
+    }.bind(this))).then(cacheResource.bind(this, method));
+};
 
-    var buildFullPath = function(base, path) {
-        return (base + path).replace(/\/{2,}/g, '/');
-    };
+var cacheResource = function(method, resource) {
+    if ('get' === method && resource.$id && resource.$expires) {
+        this._cache.put(resource.$id, resource, resource.$expires);
+    }
+    return resource;
+};
 
-    function buildQuery(options) {
-        var query = {};
+var buildFullPath = function(base, path) {
+    return (base + path).replace(/\/{2,}/g, '/');
+};
 
-        if (options && Array.isArray(options.props)) query.props = options.props.join(',');
+function buildQuery(options) {
+    var query = {};
 
-        var queryStringParts = [];
+    if (options && Array.isArray(options.props)) query.props = options.props.join(',');
 
-        for (var i in query) {
-            queryStringParts.push(encodeURIComponent(i) + '=' + encodeURIComponent(query[i]));
-        }
+    var queryStringParts = [];
```

Comment:
Not that it's a big deal, but we could probably leverage node's ```querystring``` module that will handle all of this for you and just reduces it to one line of code to return what you're looking for.  We get it for free anyway since this is built on top of node.